whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapElasticFullTextSearchService.java

```diff
@@ -90,13 +91,30 @@ protected List<RpslObject> doSearch() throws IOException {
                     return results;
                 }
 
-                private QueryBuilder getQueryBuilder(final String[] fields, final String term) {
-                    if (term.indexOf('*') == -1 && term.indexOf('?') == -1) {
-                        return new MultiMatchQueryBuilder(term, fields)
-                                .type(MultiMatchQueryBuilder.Type.PHRASE_PREFIX)
-                                .operator(Operator.AND);
+                private QueryBuilder getQueryBuilder(final String[] fields, final String term, final boolean matchExact) {
+                    if (hasWildCard()) {
```

Comment:
MatcExact  and hasWildcard seems to be contradictory. if  hasWildCard() then   createWildCardQuery() else exact match?

Reply:
I am already doing this, right?

                private QueryBuilder getQueryBuilder(final String[] fields, final String term, final boolean matchExact) {
                    if (hasWildCard()) {
                        return createWildCardQuery();
                    }

                    return matchExact ? createExactMatchQuery() :
                            new MultiMatchQueryBuilder(term, fields)
                                    .type(MultiMatchQueryBuilder.Type.PHRASE_PREFIX)
                                    .operator(Operator.AND);
                }

Reply:
Yes , That's what I am asking, why you need matchExact boolean value ? If it has wildcard , it goes to createWildCardQuery else it should be always createExactMatchQuery(). or it is for some other functionality ?

Reply:
I think if we say user needs to put wildcard, we should be consistent and get rid of matchQuery flag all together. Does RFC points that only for ips and autnums we need to do exact match  and rest can match multiple ?

Reply:
I think that the RFC only talks about ips and autnums.
Maybe discuss with Ed, I think that we discussed about it and the outcome was just to modify ips and autnums

Reply:
In that case, is it possible to check the fields here to return exactMatch or MultiMatch rather passing the boolean flag from controller level? That would make it more clearer